crates/nomination/src/lib.rs

```diff
@@ -388,16 +361,14 @@ impl<T: Config> Module<T> {
         );
         let operator = Operator::new(operator_id.clone());
         <Operators<T>>::insert(operator_id, operator.clone());
-        ext::vault_registry::set_is_nomination_operator::<T>(operator_id, true);
         Self::deposit_event(Event::<T>::NominationOptIn(operator_id.clone()));
         Ok(())
     }
 
     pub fn _opt_out_of_nomination(operator_id: &T::AccountId) -> DispatchResult {
-        let mut operator = Self::get_rich_operator_from_id(operator_id)?;
-        operator.force_refund_nominated_collateral()?;
+        let operator = Self::get_rich_operator_from_id(operator_id)?;
+        ensure!(!operator.has_nominated_collateral(), Error::<T>::HasNominatedCollateral);
```

Comment:
We actually allow Operators to force refund nominators if 1) they want to opt out (full nominator refund) or 2) want to withdraw an amount that would cause the Max Nomination Ratio to be exceeded (partial refund, so that the nominated amount does not exceed the Max Nomination Ratio)

Reply:
Okay then we would need to think of a workaround since iterating through all nominators is not really feasible.

Reply:
We can probably allow vaults to pause future nominations and force refund on a per nominator basis.

Reply:
Or perhaps push collateral that is not their own into some pool for collection.

Reply:
The force refund should be identical to a reward distribution - nominators would be able to "pull" at any time